### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by yoch**
- many code simplifications
- python 2 / 3 integration made simpler
- improve some inputs handling

Also, some bugs were fixed, I plan to write about this later.

### CODE REVIEW

**Comment by yoch** [comment]
fix #78

File: src/paho/mqtt/client.py [hunk 1]
```diff
@@ -2009,25 +2014,25 @@ def _send_connect(self, keepalive, clean_session):
...
             protocol = PROTOCOL_NAMEv311
             proto_ver = 4
+
         remaining_length = 2+len(protocol) + 1+1+2 + 2+len(self._client_id)
         connect_flags = 0
         if clean_session:
             connect_flags |= 0x02
 
         if self._will:
+            remaining_length += 2+len(self._will_topic)
```
Comment (ralight) [inline src/paho/mqtt/client.py]: This substitution is wrong.

**Comment by ralight** [comment]
This is good stuff, with a couple of minor exceptions.

Sorry to be awkward though, but could you split this out into separate commits by category of fix? I'm a simple man and like to keep things straightforward and easy to follow where possible. Something like "Replace all instances of x=x+y with x+=y"

"Replace all instances of this style of code:

x = topic
if x[0] == 0:
    ....

with

x, = topic
if x == 0:
    ...."

It's much easier to review that way.

I noticed that you've changed the behaviour a bit as well - you no longer do complete type checks on the payload data. I've not tested it, will that now automatically raise a ValueError?

**Comment by ralight** [comment]
Closing this due to PR #81.

**Pull request closed without merging**